ee/tables/execparsers/data_table/parser.go

```diff
@@ -0,0 +1,120 @@
+package data_table
+
+import (
+	"bufio"
+	"io"
+	"strings"
+)
+
+// data_table is a general parser for an input of data which conforms to columns and rows styled output.
+// Parser options
+// skipLines - The number of initial lines of data to skip. By default no lines are skipped. This can be useful if consistent undesired output/garbage is printed before the data to parse.
+// headers - The set of headers. If left blank, the parser assumes the headers are in the first line of data and splits that line to set them.
+// delimiter - The splitting string. If left blank, the parser assumes the delimiter is whitespace and uses `strings.Fields()` split method.
+type parser struct {
+	skipLines uint
+	headers   []string
+	delimiter string
+}
+
+type parserOpt func(*parser)
+
+func WithSkipLines(skipLines uint) parserOpt {
+	return func(p *parser) {
+		p.skipLines = skipLines
+	}
+}
+
+func WithHeaders(headers []string) parserOpt {
+	return func(p *parser) {
+		p.headers = headers
+	}
+}
+
+func WithDelimiter(delimiter string) parserOpt {
+	return func(p *parser) {
+		p.delimiter = delimiter
+	}
+}
+
+func NewParser(opts ...parserOpt) *parser {
+	p := &parser{
+		skipLines: 0,
+		headers:   nil,
+		delimiter: "",
+	}
+
+	for _, opt := range opts {
+		opt(p)
+	}
+
+	return p
+}
+
+func (p parser) Parse(reader io.Reader) (any, error) {
+	return p.parseLines(reader)
+}
+
+// parseLines scans a reader line by line and splits it into fields based on a delimiter.
+// The line fields are paired with a header, which is defined by an input array, or the first line of data.
+func (p parser) parseLines(reader io.Reader) ([]map[string]string, error) {
+	results := make([]map[string]string, 0)
+	scanner := bufio.NewScanner(reader)
+
+	// Skip first N lines due to provided headers or otherwise.
+	// This would likely only ever be 1 or 0, but we may want more.
+	for p.skipLines > 0 {
+		p.skipLines--
+		// Early exit if the scanner skipped past all data.
+		if !scanner.Scan() {
+			return results, nil // <-- Do we want to error here?
+		}
+	}
+
+	headerCount := len(p.headers)
+
+	for scanner.Scan() {
+		line := scanner.Text()
+
+		// headers weren't provided, so retrieve them from the first available line.
+		if headerCount == 0 {
+			p.headers = p.lineSplit(line, headerCount)
+			headerCount = len(p.headers)
+			continue
+		}
+
+		row := make(map[string]string)
+		fields := p.lineSplit(line, headerCount)
+		// It's possible we don't have the same number of fields to headers, so use
+		// min here to avoid a possible array out-of-bounds exception.
+		min := min(headerCount, len(fields))
+
+		// For each header, add the corresponding line field to the result row.
+		// Duplicate headers overwrite the set value. <-- Perhaps this should be different?
+		for c := 0; c < min; c++ {
```

Comment:
```suggestion
		// For each header, add the corresponding line field to the result row.
		// Duplicate headers overwrite the set value. <-- Perhaps this should be different?
		for c := 0; c < min(headerCount, len(fields)); c++ {
```

`i` is also the normal enumerator

Reply:
Yeah I'll switch to `i`, I sometimes just go with the first letter of purpose for the value and I put c for counters, but I can use the standard first loop enumerator.